python/uyuni/common/fileutils.py

```diff
@@ -239,13 +239,19 @@ def rhn_popen(cmd, progressCallback=None, bufferSize=16384, outputLog=None):
 
 
 def makedirs(path,  mode=int('0755', 8), user=None, group=None):
-    "makedirs function that also changes the owners"
+    """makedirs function that also changes the owners"""
 
     dirs_to_create = []
     dirname = path
 
     uid, gid = getUidGid(user, group)
 
+    if uid is None:
+        raise OSError(f"*** ERROR: user {user} doesn't exist. Cannot create path.")
```

Comment:
Code in `uyuni.common` is used on both Uyuni server and the clients, not all of which use Python 3.6+. We need to check that only code that's exclusive to Uyuni server can hit this code path or wait until we're free to be incompatible with the Python interpreters on client systems.

SUMA 4.3 will be the last release that supports the traditional stack, after its release we can start introducing incompatibilities (on purpose).